docs/rfcs/2024-02-19-settings.md

```diff
@@ -0,0 +1,60 @@
+# Migrating app settings to another device
+
+## Problem
+
+This is related to simplified database migration UX in the [previous RFC](./2024-02-12-database-migration.md).
+
+Currently, when database is imported after the onboarding is complete, users can configure the app prior to the import.
+
+Some of the settings are particularly important for privacy and security:
+- SOCKS proxy settings
+- Automatic image etc. downloads
+- Link previews
+
+With the new UX, the chat will start automatically, without giving users a chance to configure the app. That means that we have to migrate settings to a new device as well, as part of the archive.
+
+## Solution
+
+There are several possible approaches:
+- put settings to the database via the API
+- save settings as some file with cross-platform format (e.g. JSON or YAML or properties used on desktop).
+
+The second approach seems much simpler than maintaining the settings in the database.
+
+If we save a file, then there are two options:
+- native apps maintain cross-platform schemas for this file, support any JSON and parse it in a safe way (so that even invalid or incorrect JSON - e.g., array instead of object - or invalid types in some properties do not cause the failure of properties that are correct).
+- this schema and type will be maintained in the core library, that will be responsible for storing and reading the settings and passing to native UI as correct record of a given type.
+
+The downside of the second approach is that addition of any property that needs to be migrated will have to be done on any change in either of the platforms. The downside of the first approach is that neither app platform will be self-sufficient any more, and not only iOS/Android would have to take into account code, but also each other code.
```

Comment:
even with second approach iOS/Android still have to take into account each others' code, no? In a sense that settings different per OS will be making it into AppSettings type, for example different call settings, appearance settings.

How will settings that couldn't be applied (but were successfully parsed) be treated? E.g. migrating Android to iOS back to Android will preserve Android specific settings or they will be forgotten?

Reply:
yes, will be dropped